### PR TITLE
Able to open link in browser based on user preference

### LINKED ISSUE #5: [💡 FEATURE] - Open links in default browser instead of new Electron window
can you make it so that links clicked on in the document go to the default browser instead of another window?

### PR DESCRIPTION
**Pull request by mding5692**
| Status  | Type  | Env Vars Change |
| :---: | :---: | :---: |
| Ready | Feature | No |

## Problem

Resolves #5 .

## Solution

Based on these links: 
- https://github.com/electron/electron/blob/master/docs/api/shell.md#shellopenexternalurl
- https://stackoverflow.com/questions/32402327/how-can-i-force-external-links-from-browser-window-to-open-in-a-default-browser
- https://github.com/electron/electron/blob/master/docs/api/web-contents.md#event-new-window

Also adds option in store.js for user to change it based on user preference. This part probably needs some thinking or some settings window to toggle it.

## Before & After Screenshots

**BEFORE**:

Screen Shot 2020-05-22 at 11 46 47 AM

**AFTER**:

Screen Shot 2020-05-22 at 11 45 44 AM

### CODE REVIEW

**Comment by mding5692** [comment]
@alexkim205 Probably need somewhere on menu to toggle if users want to open links in browser, also whats the bug you noticed in this PR?

**Comment by alexkim205** [comment]
Agreed, do you think that option should be under the `Window` menu? 

Sorry I mislabelled!

**Comment by mding5692** [comment]
Yea sure, under window menu works, what do you think should be default value for openLinkInBrowser? true or false?

**Comment by alexkim205** [comment]
I think GDS should feel like an app isolated from the web browser, and external links should be handled by the user's browser. 

I think it should default to opening the link in the browser if it's not google affiliated.

**Comment by mding5692** [comment]
> GDS should feel like an app isolated from the web browser, and external links should be handled by the user's browser

Probably need a way to figure out which is GDS and which is external link

**Comment by mding5692** [comment]
For now for this PR I think we can just add to general window settings first an option to choose between opening links in browser or in window and then work on another PR to identify GDS links and meant-for-browser links

**Comment by mding5692** [comment]
@alexkim205 Ok added an option in Windows menu to toggle if open links in browser by default

File: src/js/menu.js [hunk 1]
```diff
@@ -97,6 +106,12 @@ const template = [
     submenu: [
       { role: "minimize" },
       { role: "zoom" },
+      {
+        label: "Open links in browser",
```
Comment (alexkim205) [inline src/js/menu.js:110]: Rename to "Open external links in browser"

File: src/js/mainwindow.js [hunk 2]
```diff
@@ -99,11 +100,17 @@ var createMainWindow = () => {
   view.webContents.on(
     "new-window",
     (event, url, frameName, disposition, options) => {
-      createChildWindow(event, url, frameName, disposition, {
-        ...options,
-        pos: win.getPosition(),
-        size: win.getSize(),
-      });
+      const shouldOpenLinkInBrowser = store.get("openLinksInBrowser");
```
Comment (alexkim205) [inline src/js/mainwindow.js:103]: check if external or google link here

**Comment by alexkim205** [comment]
I just looked over your PR and I think you organized the files really well. The PR is almost there, but we should include a check to see which is a GDS link and which is the external link. I guess I don't see the purpose of the app if the app only opened the main Google Drive window in a separate window and opened files in the user's default browser. 

Perhaps to make it more explicit, we can rename the menu item to "Open external links in browser" and add a simple regex check to differentiate links (starts with `{docs|drive}.google`?).

**Comment by mding5692** [comment]
Ok sure that works, can add that in

**Comment by mding5692** [comment]
> Perhaps to make it more explicit, we can rename the menu item to "Open external links in browser" and add a simple regex check to differentiate links (starts with `{docs|drive}.google`?).

So do all google drive-related urls begin with `{docs|drive}.google`?

**Comment by alexkim205** [comment]
I believe so. There may be edge cases that I'm unaware of, but Google Docs, Sheets, and Slides all share the docs.google prefix while the Drive page is drive.google.

**Comment by mding5692** [comment]
Ok that works then

**Comment by mding5692** [comment]
Probably needs some review and go over, also noticed the unsupported browser warning coming back

**Comment by alexkim205** [comment]
Great thanks. I had to make a hotfix for 0.3.0 because our temporary fix wasn't letting users sign in at all (see #71)

**Comment by mding5692** [comment]
> Great thanks. I had to make a hotfix for 0.3.0 because our temporary fix wasn't letting users sign in at all (see #71)

Ah ok, understand, makes sense

**Comment by alexkim205** [comment]
Hmm, I can't get external links in docs to open at all. I think we should add the same functionality that you added to the mainwindow to the childwindow.

**Comment by mding5692** [comment]
Yep, actually noticed that to while trying to open the urls in the docs

**Comment by alexkim205** [comment]
No worries, I think the newest changes should work.

**Comment by mding5692** [comment]
@alexkim205 Just added some constants for the user preferences we were setting

**Pull request merged**